lib/data-fetcher.js

```diff
@@ -157,8 +157,16 @@ function _handleFirebaseResponse(response) {
 
 function firebaseFetch(url, payload) {
   const options = _firebaseOptions(payload);
-  return fetch(url, options)
-    .then(_handleFirebaseResponse);
+  const res = fetch(url, options);
+  res.then(r => {
+    if (r.status !== 200) {
+      r.text().then(msg => {
+        // Add codebase-wide logging system
+        console.warn(`firebaseFetch error: GET ${url} => ${r.status}: ${msg}`);
+      });
+    }
+  });
```

Comment:
I think not chaining the calls make it more clear that this is an observer that does not transform the response--if you're scanning through `firebaseFetch()` to see what it does/returns, you can ignore this block, because it's not chained, and not connected to the return value.